core/src/test/java/org/infinispan/persistence/support/AsyncStoreTest.java

```diff
@@ -370,7 +370,10 @@ private static ConfigurationBuilder config(boolean passivation) {
          ConfigurationBuilder config = new ConfigurationBuilder();
          config.memory().maxCount(1).persistence().passivation(passivation).addStore(LockableStoreConfigurationBuilder.class)
                .async()
-                  .modificationQueueSize(1)
+                  // When using passivation we block a store write to key X - which in turn
+                  // gets in a deadlock as it never completes - make sure to set queue size one
+                  // higher so it can complete
```

Comment:
The ordering in passivation. The problem is the `doTestEndToEndPutPut` test delays the store write from completing in the test directly. Thus the batch can't complete and thus if you submit a new write to the store it will be enqueued in the async batch and wont' complete.

Reply:
I like the new comment :+1: